Tools/ApplicationDebugger/jacobi/README.md

```diff
@@ -0,0 +1,340 @@
+a format `[6 7 8 ... 8 7 6]^T`. Thus, the solution vector `x` has a form of `[1 1 ... 1]^T`.
+
+The program `jacobi-fixed` computes vector `x` correctly, while `jacobi-bugged`
+returns an incorrect result. With the debugger you can put a breakpoint in and 
+outside of the kernel, step through the code, and check the values of local variables. 
+Use these features to find places in `jacobi-bugged` where the program does not 
+behave as it was intended.
+
+The sample is intended for exercising the debugger, not for performance benchmarking.
+
+The debugger supports debugging kernels that run on the CPU, GPU, or
+accelerator devices.  For convenience, the `jacobi`
+code sample provides the ability to select the target device by passing the
+program `cpu`, `gpu`, or `accelerator` as the command-line argument.
+The selected device is displayed in the output.
+For the overview of the debugger, please refer to 
+[Get Started Guide](https://software.intel.com/en-us/get-started-with-debugging-dpcpp)
+of the application debugger.
+
+## Key Implementation Details
+
+The basic DPC++ implementation explained in the code includes device
+selection, buffer, accessor, and command groups.
+
+Each iteration of the Jacobi method performs the following computation:
+
+```
+x^{k+1} = D^{-1}(b - (A - D)x^k),
+```
+
+where `n x n` matrix `D` is a diagonal component of the matrix `A`.
+In the sample, this computation is offloaded to the device at each iteration.
+In the code, `x^{k+1}` corresponds to the variable `x_k1`, and
+`x^k` corresponds to `x_k`. At the end of each iteration we update `x_k` with
+the value of `x_k1`.
+
+## License
+
+Code samples are licensed under the MIT license. See
+[License.txt](https://github.com/oneapi-src/oneAPI-samples/blob/master/License.txt) for details.
+
+Third party program Licenses can be found here: [third-party-programs.txt](https://github.com/oneapi-src/oneAPI-samples/blob/master/third-party-programs.txt)
+
+## Building and Running the `jacobi` Program
+
+> Note: if you have not already done so, set up your CLI 
+> environment by sourcing  the setvars script located in 
+> the root of your oneAPI installation. 
+>
+> Linux Sudo: . /opt/intel/oneapi/setvars.sh  
+> Linux User: . ~/intel/oneapi/setvars.sh  
+> Windows: C:\Program Files(x86)\Intel\oneAPI\setvars.bat
+
+### Setup
+
+Preliminary setup steps are needed for the debugger to function.
+Please see the setup instructions in the Get Started Guide based on
+your OS:
+[Linux](https://software.intel.com/en-us/get-started-with-debugging-dpcpp-linux),
+[Windows](https://software.intel.com/en-us/get-started-with-debugging-dpcpp-windows).
+
+
+### Include Files
+
+The include folder is located at
+`%ONEAPI_ROOT%\dev-utilities\latest\include` on your development
+system.
+
+
+### Running Samples In DevCloud
+
+If running a sample in the Intel DevCloud, remember that you must
+specify the compute node (CPU, GPU, FPGA) and whether to run in
```

Comment:
I was told that the interactive mode is pretty hidden at the moment and that the recommendation is to just qsub (batch mode). The big problem with interactive is that it ties up resources for 6 hrs. Please verify the links and make sure the info is there that's needed

Reply:
@JoeOster, batch mode would practically be unusable for the debugger exercise.

Reply:
Then if it is unusable, we should have specific instruction or point to a set on how to get to an interactive prompt

Reply:
The command given right below (`$ qsub -I -l nodes=1:gpu:ppn=2`) gives the interactive prompt. Does it not suffice?

Reply:
Sorry, I don't use devcloud, so wasn't aware that the command did that. Regardless as this is a different mode than the batch mode we might want to state specifically... This command will get you into an interactive mode, allowing you to....

I am good if you don't want to point it out

Reply:
I rephrased the section a bit.